docs/source/usage_guides/deepspeed_multiple_model.md

```diff
@@ -0,0 +1,203 @@
+
+```python
+from accelerate.utils import DeepSpeedPlugin
+
+zero2_plugin = DeepSpeedPlugin(hf_ds_config="zero2_config.json")
+zero3_plugin = DeepSpeedPlugin(hf_ds_config="zero3_config.json")
+
+deepspeed_plugins = {"student": zero2_plugin, "teacher": zero3_plugin, }
+```
+
+When doing so,  `zero2_config.json` should be configured for full training (so specifying `scheduler` and `optimizer` if not utilizing your own) while `zero3_config.json` should *only* be configured for the model,
+such as:
+
+```json
+{
+    "bf16": {
+        "enabled": "auto"
+    },
+    "zero_optimization": {
+        "stage": 3,
+        "offload_param": {
+            "device": "cpu",
+            "pin_memory": true
+        },
+        "overlap_comm": true,
+        "sub_group_size": 1e9,
+        "reduce_bucket_size": 1e9,
+        "stage3_prefetch_bucket_size": 1e9,
+        "stage3_param_persistence_threshold": 1e9,
+        "stage3_max_live_parameters": 1e9,
+        "stage3_max_reuse_distance": 1e9,
+        "stage3_gather_16bit_weights_on_model_save": true
+    },
+    "train_micro_batch_size_per_gpu": 1
+}
+```
+
+<Tip>
+
+    DeepSpeed will raise an error if we don't specify a `train_micro_batch_size_per_gpu`, despite not really 
+    training this particular model
+
+</Tip>
+
+From here, we can create a single [`Accelerator`] and pass in both configurations:
+
+```python
+from accelerate import Accelerator
+
+accelerator = Accelerator(deepspeed_plugin=deepspeed_plugins)
+```
+
+Now let's see how to use them.
+
+### The student model
+
+By default, `accelerate` will set the first item in the `dict` as the default/enabled plugin (so here our `"student"` plugin). We can verify this by using the [`utils.deepspeed.get_active_deepspeed_plugin]` function to see which one is being used:
+
+```python
+active_plugin = get_active_deepspeed_plugin(accelerator.state)
+assert active_plugin is deepspeed_plugins["student"]
+```
+
+Since the `student` is the currently active plugin, let's go ahead and prepare our model, optimizer, and scheduler like normal:
+
+```python
+student_model, optimizer, scheduler = ...
+student_model, optimizer, scheduler, train_dataloader = accelerator.prepare(student_model, optimizer, scheduler, train_dataloader)
+```
+
+From here, we need to deal with the teacher model.
+
+### The teacher model
+
+First we need to tell the `Accelerator` that the zero3/teacher ZeRO configuration should be used:
+
+```python
+accelerator.state.enable_deepspeed_plugin("teacher")
+```
+
+Doing so will disable the `"student"` plugin and enable the `"teacher"` one instead. This will update the
+DeepSpeed stateful config inside of `transformers`, and change which plugin configuration gets called when using
```

Comment:
It's not clear how `transformers` comes into play here. What if the user is not using transformers? It should still work, right?

Reply:
Yes and no. If not using `transformers` they need to manually call `ZeroInit`, which is what this is talking about